samples/snippets/user_credentials.py

```diff
@@ -23,26 +23,16 @@
 import argparse
 
 
-def main(project, launch_browser=True):
+def main(project):
     # [START bigquery_auth_user_flow]
     from google_auth_oauthlib import flow
 
-    # TODO: Uncomment the line below to set the `launch_browser` variable.
-    # launch_browser = True
-    #
-    # The `launch_browser` boolean variable indicates if a local server is used
-    # as the callback URL in the auth flow. A value of `True` is recommended,
-    # but a local server does not work if accessing the application remotely,
-    # such as over SSH or from a remote Jupyter notebook.
```

Comment:
gcloud allows you to complete auth on a remote machine with no browser `gcloud auth application-default login --no-browser`, available in >=372.0.0. 

[This doc](https://docs.google.com/document/d/1YWiIiZiX3Ps7gcJkLBf49cbM6vqtsD1LeO39yyN0KW8/edit#) has an overview of what the flow looks like. I believe this new flag will take care of the general 'accessing the application remotely case' as long as the user can run gcloud on the target machine.

Reply:
@tswast @shollyman Are there cases where a user might not be able to run `gcloud` on the machine where the notebook is hosted?

Reply:
> @tswast @shollyman[](https://teams.googleplex.com/shollyman@google.com) Are there cases where a user might not be able to run gcloud on the machine where the notebook is hosted?

Oh dear. 😱  I'm not sure about "can't" but this will add significant friction to folks using hosted notebooks on other clouds such as Azure Machine Learning Notebooks or Amazon SageMaker.

Reply:
Thanks for bringing this to my attention. I believe I'll need to prioritize https://github.com/pydata/pydata-google-auth/issues/53 and ideally https://github.com/pydata/pydata-google-auth/issues/48 as well due to this auth change.

Reply:
Kubeflow notebooks and other self-hosted notebook systems like Jupyter Hub will also have a problem.

Reply:
Thanks Tim, it looked like some other teams working with Notebooks are also trying to figure out how this impacts their users. dujinhui@ is on the team leading the OOB deprecation if you need a point of contact.